base/utils/tests/testthat/test.utils.R

```diff
@@ -69,13 +69,30 @@ test_that("summarize.result works appropriately", {
                            specie_id = 1,
                            n = 1,
                            mean = sqrt(1:10),
-                           stat = 'none',
-                           statname = 'none'
+                           stat = NA,
+                           statname = NA
                            )
+  # check that individual means produced for distinct sites
+  expect_that(summarize.result(testresult)$mean, equals(testresult$mean)) 
+  
+  # check that four means are produced for a single site
   testresult2 <- transform(testresult, site_id= 1) 
-  expect_that(summarize.result(testresult)$mean, equals(testresult$mean))
-  expect_that(nrow(summarize.result(testresult2)), equals(4))
-})
+  expect_that(nrow(summarize.result(testresult2)), equals(4))  
+  
+  # check that if stat == NA, SE will be computed
+  testresult3 <- summarize.result(testresult2)
+  expect_true(all(!is.na(testresult3$stat)))
+  expect_equal(testresult3$n, c(3L, 2L, 2L, 3L))
+  expect_equal(round(testresult3$stat, 3), c(0.359, 0.177, 0.293, 0.206))
+  expect_equal(round(testresult3$mean, 3), c(1.656, 2.823, 1.707, 2.813))
+  
+  # check that site groups correctly for length(site) > 1
+  testresult4 <- rbind.data.frame(testresult2, transform(testresult2, site_id= 2))
+  testresult5 <- summarize.result(testresult4)
+  expect_true(all(!is.na(testresult5$stat)))
+  expect_equal(nrow(testresult5), 8)
+  
+  })
```

Comment:
Thank you for including unit tests! We've been, uh, inconsistent about them recently.